### DISCUSSION TITLE
Print current option help

### DISCUSSION THREAD

**Discussion by tobiasBora** [marlonrichert/zsh-autocomplete#722] [Q&A]
## What do you want?

Sometimes, I'm not sure whether I'm typing the good option, say that I don't remember what `ls -X` does exactly. I would like to read the doc in the first entry of the help menu… unfortunately the typed option does not appear at all in the help menu:

![image](https://github.com/marlonrichert/zsh-autocomplete/assets/2164118/724ab4ea-a1e0-473c-9ae5-65b6798e9187)

Similarly, I'm not sure to understand why, but the list of options that is printed if I just type `-` is much smaller than the list of options that I can get if I type first a dummy entry like `-A`:

![Peek 27-03-2024 20-37](https://github.com/marlonrichert/zsh-autocomplete/assets/2164118/d3e25251-1e21-45df-91ec-a5ee95d7af40)

## Why do you want this?

Poor memory?

## Who else would benefit from this?

I guess many people.

## How should it work?

When I type `-`, it should print all available options (like what happens with the MORE thing when I typed a character), and if I type `-X` or `--someoptions`, the option I just typed should appear first in the `option` list so that I can read its documentation.

**Comment by marlonrichert** [comment]
First off, this is not handled by Autocomplete. It's done by Zsh itself.

Second, I tried a couple of things, but I could not get it to work the way you want it. Please ask for more help on the official [Zsh users mailing list](https://zsh.sourceforge.io/Arc/mlist.html).
